### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by bpoldrack**
Following up on PR #44, this catches `HTTPError` from authentication request explicitly and tries to give a more meaningful message, instead of something like the rather generic ones coming from the server like "Client Error for url xxx" when providing invalid credentials.

Examples for invalid credentials showing result dict and its rendering:

```
❱ datalad -f json_pp xnat-init https://xnat.kube.fz-juelich.de                                                                                       
{
  "action": "xnat_init",
  "exception": "RuntimeError(Failed to access the XNAT server. Reason: unauthorized)",
  "exception_traceback": "[init.py:__call__:105,platform.py:__init__:104,platform.py:__init__:95,models.py:raise_for_status:953]",
  "message": [
    "%s",
    "Failed to access the XNAT server. Reason: unauthorized"
  ],
  "path": "/tmp/some",
  "refds": "/tmp/some",
  "status": "error",
  "type": "dataset"
}
```

```
❱ datalad xnat-init https://xnat.kube.fz-juelich.de                                                                                                  1 !
xnat_init(error): . (dataset) [Failed to access the XNAT server. Reason: unauthorized]
```

### CODE REVIEW

File: datalad_xnat/platform.py [hunk 1]
```diff
@@ -82,15 +87,23 @@ def __init__(self, url, credential):
...
         try:
             session.post(self._get_api('session_token')).raise_for_status()
+        except HTTPError as e:
+            # TODO: Accessing non-public requests.status_codes._codes isn't nice.
+            #       But the respective LookupDict seems only useful to look up a
+            #       code based on a name, not the other way around. That is,
+            #       however, what we need here to be more informative than
+            #       XNAT's "Client Error for url XXX" for an 401 for example.
+            msg = e.response.reason or \
+                  status_codes._codes[e.response.status_code][0]
```
Comment (mih) [inline datalad_xnat/platform.py:101]: I think it would be better to use https://docs.python.org/3/library/http.html#http.HTTPStatus

**Review by mih — CHANGES_REQUESTED** [review]
Thx, small comment to avoid private access.

File: datalad_xnat/platform.py (hunk 1 above)
Comment (mih) [reply datalad_xnat/platform.py:101]: `{i.value: i.phrase for i in http.HTTPStatus}`

Comment (bpoldrack) [reply datalad_xnat/platform.py]: Done. Also looked in `.description` as an alternative, but that phrasing requires "protocol awareness" - not more useful for users.

**Review by mih — APPROVED** [review]
Thx

**Comment by adswa** [comment]
@all-contributors please add @bpoldrack for code

**Pull request merged**